lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -238,50 +230,10 @@
         }
     });
 
-    platform.registerChakraLibraryFunction("DefaultStringSortCompare", function(left, right) {
-        // this version is used when the array was already strings
-        // as the sort only ever checks for < 0 on the return value of compare functions
-        // only have to handle this case
-        if (left < right) {
-            return -1;
-        }
-        return 0;
-    });
-
     platform.registerChakraLibraryFunction("DefaultSortCompare", function(left, right) {
         // as the sort only ever checks for < 0 on the return value of compare functions
         // only have to handle this case
-        if (left.string < right.string) {
-            return -1;
-        }
-        return 0;
-    });
-
-    platform.registerChakraLibraryFunction("CreateCompareArray", function(array, length) {
-        let useCompareArray = false;
-        let i = 0;
-        while (i < length) {
-            if (typeof array[i++] !== "string") {
-                useCompareArray = true;
-                break;
-            }
-        }
-
-        if (useCompareArray === true) {
-            const compArray = [];
-            compArray.__proto__ = null;
-            i = 0;
-            let value;
-            while (i < length) {
-                value = array[i];
-                compArray[i++] = {
-                    value : value,
-                    string : "" + value
-                };
-            }
-            return compArray;
-        }
-        return array;
+        return ("" + left) < ("" + right) ? -1 : 0;
```

Comment:
Thought doing this as a single line was neater than the `if` I used before - string coercion (per spec) is done by appending to an empty string.